server/src/collection.rs

```diff
@@ -157,6 +157,29 @@ impl Collection {
         self.alias.lock().await.clone()
     }
 
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let items = self.items.lock().await;
+
+        for item in items.iter() {
+            if !item.is_locked().await {
+                item.set_locked(true).await?;
+            }
+            let manager = self.manager.lock().await;
```

Comment:
Why you would have to get the manager multiple times ;)

Reply:
Yeah I know :) Since the Item side locks the manager first, this was the most suitable place to lock/get the manager.

Reply:
hmm, what?